libmscore/rendermidi.cpp

```diff
@@ -1009,6 +1009,8 @@ void Score::updateHairpin(Hairpin* h)
       Fraction tick  = h->tick();
 
       Segment* seg = st->score()->tick2segment(tick);
```

Comment:
Maybe it would be more appropriate to use [`h->startSegment()`](https://github.com/musescore/MuseScore/blob/551b9d93dc28bd8ce94d532c427c585d122d560a/libmscore/spanner.cpp#L781) here which would find the relevant segment (by calling `tick2rightSegment` internally) so the function could do something meaningful here instead of just returning. Then the check below may be not needed.

Reply:
Sounds good, but what do you mean by something meaningful?

Reply:
Exactly what it would do otherwise if the hairpin would start on that right segment :) A hairpin spans for some range that probably exceeds the length of one segment so in case the hairpin does not start exactly at some segment applying it at least for the available segment range sounds more correct than not applying it at all. I just propose to replace `tick2segment()` with `h->startSegment()` so that the relevant segment probably always gets found.

Reply:
I didn't realise it was possible to get a hairpin to start not exactly on a segment, how would you go about doing that?

Reply:
(I have made your suggested change, btw)

Reply:
> I didn't realise it was possible to get a hairpin to start not exactly on a segment, how would you go about doing that?

Actually I believe this shouldn't normally happen but according to [this comment](https://musescore.org/en/node/287615#comment-911319) this hairpin was placed there intentionally. Still I am not sure whether this actually works correctly.